components/sup-protocol/protocols/ctl.proto

```diff
@@ -82,7 +82,7 @@ message SvcLoad {
   // Remote http URL for the Builder service to receive package updates from.
   optional string bldr_url = 6;
   // Remote channel on the Builder service to receive package updates from.
-  optional string bldr_channel = 7;
+  optional sup.types.ChannelIdent bldr_channel = 7;
```

Comment:
Hrmm... I'm not sure that we can it this way. The [docs](https://developers.google.com/protocol-buffers/docs/proto#updating) imply that you can change strings to messages if the string contains an encoding of the message, which I don't think is the case here.

I think you might run into issues if your `hab-sup` and `hab` binaries aren't in lockstep.

You might be able to have a new field, called simply `channel`, and then handle the fallback conversion logic when the protobuf is consumed.

If this gets too hairy, this might be better as a separate PR.

Reply:
That sounds like a rather large amount of effort for negligible returns. The big thing motivating the change initially is wrapping the `String` so that we can have a channel-specific `Default`. However, the necessity of another protobuf-specific type to also wrap the `String`, but which we *can't` make have a sensible default seems potentially more confusing. Maybe it would be different if we had `ChannelIdent` when the .proto was created, but as is, I think we should leave it as `string bldr_channel` and do the conversion when getting data in and out of `SvcLoad`. That seems to me just as valid a frontier.

Reply:
Sounds good... as long as we don't break the protocol, and the logic for going from Protobuf into our internal types (and vice versa) is centralized, I'm 👍

Reply:
https://github.com/habitat-sh/habitat/pull/6074/commits/5e6d7914111baa0b7b4a9917b1a10d280b8b8343 should remove this